### PR TITLE
Add Net.WalkMatch API

### PR DESCRIPTION
**Pull request by fujita**
For iterating all the routes that match a given route.

### CODE REVIEW

**Comment by k-sone** [comment]
Sorry for responding late.

Sample Code:
```go
package main

import (
        "fmt"
        "net"

        "github.com/k-sone/critbitgo"
)

func testWalkMatch(trie *critbitgo.Net, n string) {
        f := func(n *net.IPNet, _ interface{}) bool {
                fmt.Println(n)
                return true
        }

        fmt.Printf("Case of %s\n", n)
        _, s, _ := net.ParseCIDR(n)
        trie.WalkMatch(s, f)
}

func main() {
        trie := critbitgo.NewNet()
        trie.AddCIDR("10.0.0.0/8", nil)
        trie.AddCIDR("192.168.0.0/16", nil)
        trie.AddCIDR("192.168.1.0/24", nil)
        trie.AddCIDR("192.168.1.0/28", nil)
        trie.AddCIDR("192.168.2.0/24", nil)

        testWalkMatch(trie, "192.168.1.1/32")
}
```
Output:
```
Case of 192.168.1.1/32
192.168.0.0/16
192.168.1.0/24
```

"192.168.1.0/28" is missing.

File: net.go [hunk 1]
```diff
@@ -203,6 +203,47 @@ func (n *Net) Walk(r *net.IPNet, handle func(*net.IPNet, interface{}) bool) {
 	})
 }
 
+func walkMatch(p *node, key []byte, handle func(*net.IPNet, interface{}) bool) bool {
+	if p.internal != nil {
+		if !walkMatch(&p.internal.child[0], key, handle) {
+			return false
+		}
+
+		if key[p.internal.offset]&p.internal.bit > 0 {
```
Comment (k-sone) [inline net.go]:
```diff
-		if key[p.internal.offset]&p.internal.bit > 0 {
+		if p.internal.offset >= len(key)-1 || key[p.internal.offset]&p.internal.bit > 0 {
```

Comment (fujita) [reply net.go]: Oops, thanks! Fixed.
Can you want me to rebase this branch against the current master so that you can merge this without conflicts?

**Comment by k-sone** [comment]
Thanks!

**Pull request merged**